### PR TITLE
Improve docstrings, add API section to Sphinx

### PR DESCRIPTION
**Pull request by cmarqu**

### CODE REVIEW

File: junitparser/junitparser.py [hunk 1]
```diff
@@ -647,15 +643,15 @@ def write(self, filepath: str = None, pretty=False):
 class JUnitXml(Element):
     """The JUnitXml root object.
 
-    It may contains a :class:`TestSuites` or a :class:`TestSuite`.
+    It may contain :class:`TestSuites` or a :class:`TestSuite`.
```
Comment (cmarqu) [inline junitparser/junitparser.py:646]: Actually, there is no TestSuites class (anymore?).

Comment (EnricoMi) [reply junitparser/junitparser.py:646]: I think this note refers to `<TestSuites>` and `<TestSuite>` tags that can be the root element of the XML.

File: junitparser/junitparser.py [hunk 2]
```diff
@@ -152,27 +153,27 @@ def __repr__(self):
...
-        """Construct Junit objects from a XML string."""
+        """Construct JUnit object *cls* from XML string *test*."""
         instance = cls()
         instance._elem = etree.fromstring(text)  # nosec
         return instance
 
     @classmethod
     def fromelem(cls, elem):
-        """Constructs Junit objects from an elementTree element."""
+        """Construct JUnit objects from an ElementTree element *elem*."""
```
Comment (EnricoMi) [inline junitparser/junitparser.py:176]:
```suggestion
        """Construct JUnit object *cls* from an ElementTree element *elem*."""
```

**Review by EnricoMi — APPROVED** [review]
Nice clean-up of the docstrings!

**Pull request merged**